src/main/java/org/iitr/muzo/controller/MainMuzoController.java

```diff
@@ -0,0 +1,64 @@
+package org.iitr.muzo.controller;
+
+import org.iitr.muzo.dao.TeacherDao;
+import org.iitr.muzo.models.Teacher;
+import org.iitr.muzo.models.UserDetails;
+import org.iitr.muzo.services.PlaylistService;
+import org.iitr.muzo.services.TeacherService;
+import org.iitr.muzo.services.UserDetailService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+@RestController
+@RequestMapping(path = "/muzo")
+public class MainMuzoController {
+
+//    private TeacherDao teacherDao;
+
+    private UserDetailService userDetailsService;
+    private PlaylistService playlistService;
+
+    @Autowired
+    public MainMuzoController(UserDetailService userDetailsService, PlaylistService playlistService){
+        this.userDetailsService = userDetailsService;
+        this.playlistService = playlistService;
+    }
+
+    @GetMapping(path = "/getAllUsers")
+    public ResponseEntity<Iterable<UserDetails>> getAllUsers(){
+        return new ResponseEntity<Iterable<UserDetails>>( userDetailsService.getAllUsers(), HttpStatus.OK);
+    }
+
+    @PostMapping(path = "/saveUser")
+    public ResponseEntity saveUser(@RequestParam String username, @RequestParam String password,
+                           @RequestParam String email, @RequestParam String name, @RequestParam String phone,
+                           @RequestParam String dob_string) throws ParseException {
```

Comment:
What happens if a paramater is missing is it throwing parse exception?

Reply:
Parse exception is for the String to date conversion, not for missing param. I dont think it will go through if a parameter is missing. (you probably need atleast a null/ empty value)